Add num-executions to the "dags next_execution" cli command

mdediana opened the pull request:
Add `--num-executions` option to the `dags next_execution` sub-command. If the option is not present it defaults to 1, which is the command's current behavior. This option is particularly useful when analyzing irregular schedules, such as some resulting from Cron expressions.

The screenshot below shows the result of using `--num-executions 5` on a schedule generated by `0 2,3,10 * * *`.

![airflow_next_execution_n_5](https://user-images.githubusercontent.com/30055/83327792-e6b81e00-a254-11ea-8768-f67835271684.png)

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

mdediana committed:
Add num-executions to next_execution sub-command

The "next_execution" cli sub-command now accepts an optional number of
executions to be returned. This is particularly useful for checking
non-regular schedule intervals, such as those created by some cron
expressions.

mik-laj commented on airflow/cli/commands/dag_command.py:
```diff
@@ -277,15 +277,24 @@ def dag_next_execution(args):
 
         if next_execution_dttm is None:
             print("[WARN] No following schedule can be found. " +
                   "This DAG may have schedule interval '@once' or `None`.")
```
```suggestion
                  "This DAG may have schedule interval '@once' or `None`.", file=sys.stderr)
```

mik-laj commented on airflow/cli/commands/dag_command.py:
```diff
@@ -277,15 +277,24 @@ def dag_next_execution(args):
+                next_execution_dttm = dag.following_schedule(next_execution_dttm)
+                print(next_execution_dttm)
     else:
         print("[WARN] Only applicable when there is execution record found for the DAG.")
```
```suggestion
        print("[WARN] Only applicable when there is execution record found for the DAG.", file=sys.stderr)
```

mik-laj commented on tests/cli/commands/test_dag_command.py:
```diff
@@ -249,36 +248,41 @@ def test_next_execution(self):
             # It prints `None` in such cases
             self.assertIn("None", out)
 
+        # Test invalid num-executions
```
Should we return non-zero exit code on invalid input?

mik-laj commented on airflow/cli/cli_parser.py:
```diff
@@ -182,6 +182,13 @@ def add_to_parser(self, parser: argparse.ArgumentParser):
+ARG_NUM_EXECUTIONS = Arg(
+    ("-n", "--num-executions"),
+    default=1,
+    type=int,
```
Should we not prohibit negative or zero numbers? https://stackoverflow.com/a/25296177

mdediana committed:
Use stderr instead of stdout

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

mdediana committed:
Use stderr instead of stdout

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

mik-laj commented on airflow/cli/commands/dag_command.py:
```diff
@@ -277,17 +277,26 @@ def dag_next_execution(args):
     if dag.get_is_paused():
         print("[INFO] Please be reminded this DAG is PAUSED now.")
```
```suggestion
        print("[INFO] Please be reminded this DAG is PAUSED now.", file=sys.stderr)
```

mdediana committed:
Use stderr instead of stdout

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

mdediana committed:
Use positive integer as type

mdediana committed:
Rename positive int check

Use positive_int instead of positive_int_type to be consistent with
other types.

mdediana committed:
Change test for int type

If ValueError exception is not caught, argparse prints a message like
"invalid positive_int value: 'a'" for non ints, while the messaging for
ints which are not positive is invalid positive int value: '-1' (without
underscore).

mik-laj reviewed (approved):
Fantastic .I look forward to your next contributions. Do you have any plans?
